app/src/services/conditional_token.ts

```diff
@@ -27,10 +27,14 @@ class ConditionalTokenService {
   provider: any
 
   constructor(address: string, provider: any, signerAddress: string) {
-    const signer: Wallet = provider.getSigner()
-
-    this.contract = new ethers.Contract(address, conditionalTokensAbi, provider).connect(signer)
+    try {
```

Comment:
I don't like this. There has to be a better way. Either checking if there's an account present in the contract, or using one of the constructor parameters to indicate if a signer is present. In fact, there's a `signerAddress` parameter, so maybe that can be used?

Reply:
Done!